nilearn/_utils/testing.py

```diff
@@ -60,6 +61,43 @@ def assert_warns(warning_class, func, *args, **kw):
         return output
 
 
+# we use memory_profiler library for memory consumption checks
+try:
+    from memory_profiler import memory_usage
+
+    def with_memory_usage(func):
+        """A decorator to skip tests requiring memory_profiler."""
+        return func
+
+    def memory_used(func, *args, **kwargs):
+        """Compute memory usage when executing func."""
+        gc.collect()
+        mem_use = memory_usage((func, args, kwargs),
+                               interval=.001, timeout=1)
+        return max(mem_use) - min(mem_use)
+
+except ImportError:
+    def with_memory_usage(func):
+        """A decorator to skip tests requiring memory_profiler."""
+        def dummy_func():
+            raise nose.SkipTest('Test requires memory_profiler.')
+        return dummy_func
+
+    def memory_used(func, *args, **kwargs):
+        """Return an invalid value."""
+        return -1
+    memory_usage = None
+
+
+def assert_memory_less_than(expected_value, tolerance,
+                            callable_obj=None, *args, **kwargs):
+    if expected_value < 1:  # MiB
```

Comment:
> Given the %memit behaviour we saw, I would feel a lot more confident to check that the memory usage is not smaller than 10 MiB

I added an extra check one the memory is measured. You were right : with 10MiB objects, the measured memory was 0.0... It works as expected with larger objects.